src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -281,44 +278,35 @@ private CaptureGitMetadataAction(ProviderFactory providers, CustomValueSearchLin
-            if (!isGitInstalled()) {
-                return;
-            }
-
-            String gitRepo = execAndGetStdOut("git", "config", "--get", "remote.origin.url");
-            String gitCommitId = execAndGetStdOut("git", "rev-parse", "--verify", "HEAD");
-            String gitCommitShortId = execAndGetStdOut("git", "rev-parse", "--short=8", "--verify", "HEAD");
-            String gitBranchName = getGitBranchName(() -> execAndGetStdOut("git", "rev-parse", "--abbrev-ref", "HEAD"));
-            String gitStatus = execAndGetStdOut("git", "status", "--porcelain");
-
-            if (isNotEmpty(gitRepo)) {
-                buildScan.value("Git repository", redactUserInfo(gitRepo));
-            }
-            if (isNotEmpty(gitCommitId)) {
-                buildScan.value("Git commit id", gitCommitId);
-            }
-            if (isNotEmpty(gitCommitShortId)) {
-                customValueSearchLinker.addCustomValueAndSearchLink("Git commit id", "Git commit id short", gitCommitShortId);
-            }
-            if (isNotEmpty(gitBranchName)) {
-                buildScan.tag(gitBranchName);
-                buildScan.value("Git branch", gitBranchName);
-            }
-            if (isNotEmpty(gitStatus)) {
+            boolean isGitInstalled = isGitInstalled();
+            Optional<String> gitRepo = given(isGitInstalled).fromCmd("git", "config", "--get", "remote.origin.url").resolve();
+            Optional<String> gitCommitId = given(isGitInstalled).fromCmd("git", "rev-parse", "--verify", "HEAD").resolve();
+            Optional<String> gitCommitShortId = given(isGitInstalled).fromCmd("git", "rev-parse", "--short=8", "--verify", "HEAD").resolve();
+            Optional<String> gitBranchName = given(isGitInstalled).fromCmd("git", "rev-parse", "--abbrev-ref", "HEAD").fromEnv(this::getGitBranchNameFromEnv).resolve();
+            Optional<String> gitStatus = given(isGitInstalled).fromCmd( "git", "status", "--porcelain").resolve();
+
+            gitRepo.ifPresent(s -> buildScan.value("Git repository", redactUserInfo(s)));
+            gitCommitId.ifPresent(s -> buildScan.value("Git commit id", s));
+            gitCommitShortId.ifPresent(s -> customValueSearchLinker.addCustomValueAndSearchLink("Git commit id", "Git commit id short", s));
+            gitBranchName.ifPresent(s -> {
+                buildScan.tag(s);
+                buildScan.value("Git branch", s);
+            });
+            gitStatus.ifPresent(s -> {
                 buildScan.tag("Dirty");
-                buildScan.value("Git status", gitStatus);
-            }
+                buildScan.value("Git status", s);
+            });
 
-            if (isNotEmpty(gitRepo) && isNotEmpty(gitCommitId)) {
-                if (gitRepo.contains("github.com/") || gitRepo.contains("github.com:")) {
-                    Matcher matcher = Pattern.compile("(.*)github\\.com[/|:](.*)").matcher(gitRepo);
+            if (gitRepo.isPresent() && gitCommitId.isPresent()) {
```

Comment:
the previous logic would also ensure the resolved value is not empty, could we also check that here?

Reply:
I could but the current code implicitly already handles the empty strings in the nested ifs

Reply:
I've hardened this by making sure having an empty string does not happen (see GitMetadataResolver logic and test)